projects/plugins/boost/app/features/optimizations/Optimizations.php

```diff
@@ -103,7 +103,8 @@ public function setup() {
 				continue;
 			}
 
-			$optimization->feature->setup();
+			add_action( $optimization->feature->setup_trigger(), array( $optimization->feature, 'setup' ) );
+
 			$this->register_endpoints( $optimization->feature );
```

Comment:
This probably also isn't doing anything. It looks like the endpoints are being registered in the features individually. And the `Optimizations::register_endpoints` function doesn't do any work. If I am right about this, perhaps it would be good also remove this call and function implementation as this refactor would be relevant to your PR.

Reply:
Thanks! I hadn't noticed this. I actually updated optimization endpoint registration to happen inside Optimizations->setup as that seems to make the most sense. Done with 65c0bef.

Though on one hand we could remove the Has_Endpoints contract and leave features decide how to register endpoints. On the other - simply declaring a list of endpoints and leaving the main class to handle registration makes the most sense (and saves 1 line per feature 😅).